frontend/state_management/Slices/MenuSlice.ts

```diff
@@ -0,0 +1,36 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { MenuState } from '../Reducers/MenuReducers';
```

Comment:
Please just move the interface type from the old reducer into your new slice so we can delete old files without issues later.